docs/topics/signals.rst

```diff
@@ -52,9 +51,54 @@ Deferred signal handlers
 ========================
 
 Some signals support returning :class:`~twisted.internet.defer.Deferred`
-objects from their handlers, see the :ref:`topics-signals-ref` below to know
-which ones.
+objects from their handlers, allowing you to run asynchronous code that
+does not block Scrapy. If a signal handler returns a
+:class:`~twisted.internet.defer.Deferred`, Scrapy waits for that
+:class:`~twisted.internet.defer.Deferred` to fire.
 
+Let's take an example::
+
+    class SignalSpider(scrapy.Spider):
+        name = 'signals'
+        start_urls = ['http://quotes.toscrape.com/page/1/']
+
+        @classmethod
+        def from_crawler(cls, crawler, *args, **kwargs):
+            spider = super(SignalSpider, cls).from_crawler(crawler, *args, **kwargs)
+            crawler.signals.connect(spider.item_scraped, signal=signals.item_scraped)
+            return spider
+
+        def item_scraped(self, item):
+            # Send the scraped item to the server
+            d = treq.post(
+                'http://localhost:3000/add',
+                json.dumps({
+                    'text': item['text'],
+                    'author': item['author'],
+                    'tags': item['tags'],
+                }).encode('ascii'),
+                headers={b'Content-Type': [b'application/json']}
+            )
+            d.addCallback(print_response)
```

Comment:
This line might be confusing. `print_response` is not defined, but since it's not necessary for the example (as it's probably just to see the effects of the request) I'd say it'd be best to remove it.

Reply:
@elacuesta Thanks for your review. I have removed the line in c9285431006dd03391dcadebf833db11124b61df